coops-ui/src/components/AnnualReport/RegisteredOfficeAddress.vue

```diff
@@ -285,38 +304,42 @@ export default class RegisteredOfficeAddress extends Vue {
   private resetAddress (): void {
     this.deliveryAddress = { ...this.deliveryAddressOriginal }
     this.mailingAddress = { ...this.mailingAddressOriginal }
+    this.emitAddresses()
     this.emitModified()
   }
 
   /**
    * Loads the office addresses from an API call.
-   *
-   * @param legalEntityNumber the identifier for the legal entity whose office addresses are fetched.
    */
   private loadAddressesFromApi (legalEntityNumber: string): void {
     if (legalEntityNumber) {
       const url = legalEntityNumber + '/addresses'
       axios.get(url)
         .then(response => {
           if (response && response.data) {
-            if (response.data.deliveryAddress) {
-              this.deliveryAddressOriginal = { ...response.data.deliveryAddress }
-              this.deliveryAddress = { ...response.data.deliveryAddress }
+            const deliveryAddress = response.data.deliveryAddress
+            if (deliveryAddress) {
+              this.deliveryAddressOriginal = { ...deliveryAddress }
+              // If parent page loaded draft before this API call, don't overwrite draft delivery address.
+              // Otherwise this API call finished before parent page loaded draft, or parent page won't
+              // load draft (ie, this is a new filing) -> initialize delivery address.
+              if (isEmpty(this.deliveryAddress)) this.deliveryAddress = { ...deliveryAddress }
             } else {
-              console.log('loadAddressesFromApi() error - invalid Delivery Address')
-              console.log(response.data.deliveryAddress)
+              console.log('loadAddressesFromApi() error - invalid Delivery Address =', deliveryAddress)
```

Comment:
Nice - I just learned this recently, so much better and still able to inspect the object in the browser console!